docs-internal/rsc-architecture-deep-dive/01-ror-pro-rsc-request-flow.md

```
@@ -0,0 +1,371 @@
+the raw bytes after the `\n`, length‑prefixed so it never needs escaping. This is the whole
+point of the framing: keeping the bulk of the payload out of `JSON.stringify` avoids ~30%
+escaping overhead (`packages/react-on-rails-pro/src/streamingUtils.ts`). The metadata is a
+small JSON object built by `buildRenderMetadata` (in `react-on-rails/serverRenderUtils`),
+e.g.
+`{"consoleReplayScript":"…","clientProps":{…},"hasErrors":false,"renderingError":null,"isShellReady":true,"payloadType":"string"}`
+(`payloadType` is appended in `streamingUtils.ts`). Note there is **no `html` field**.
+
+`injectRSCPayload` maintains **four ordered buffers** and flushes them in a strict order so the
+browser never tries to use a payload that hasn't arrived:
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)